src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const step = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          clone[key] = actions[i].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+          delete clone[actions[i].keysToRemove[y]];
+        }
```

Comment:
You can use for-of here
```suggestion
        for (const key of actions[i].keysToRemove) {
          delete clone[key];
        }
```